django/db/models/functions/comparison.py

```diff
@@ -74,6 +74,23 @@ def as_oracle(self, compiler, connection, **extra_context):
         return self.as_sql(compiler, connection, **extra_context)
 
 
+class Collate(Func):
+    function = 'COLLATE'
+    template = '%(expressions)s %(function)s %(collation)s'
+
+    def __init__(self, *expressions, **extra):
+        if len(expressions) != 2:
+            raise ValueError('Collate must take exactly two arguments')
+        collation = expressions[-1]
+        expressions = [expressions[0]]
+        super().__init__(*expressions, **extra)
+        self.extra['collation'] = collation
+
+    def as_sql(self, compiler, connection, **extra_context):
+        self.extra['collation'] = connection.ops.quote_name(self.extra['collation'])
```

Comment:
We try to avoid altering expressions during the compilation phase as it can lead to hard to diagnose issues what about

```suggestion
        self.collation = collation

    def as_sql(self, compiler, connection, **extra_context):
        extra_context.setdefault('collation', connection.ops.quote_name(self.collation)
```